src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+
+#include "compiled_model.hpp"
+
+namespace ov {
+namespace npuw {
+
+class LLMCompiledModel : public ov::npuw::ICompiledModel {
+    using GetPropertiesMap =
+        std::map<std::string, std::tuple<ov::PropertyMutability, std::function<ov::Any(const ::intel_npu::Config&)>>>;
+public:
+    struct KVCacheDesc {
+        uint32_t max_prompt_size;
+        uint32_t total_size;
+        uint32_t num_stored_tokens;
+        uint32_t dim;
+    };
+
+    LLMCompiledModel(const std::shared_ptr<ov::Model>& model,
+                     const std::shared_ptr<const ov::IPlugin>& plugin,
+                     const ov::AnyMap& properties);
+    void export_model(std::ostream& model) const override;
+    std::shared_ptr<const ov::Model> get_runtime_model() const override;
+
+    void set_property(const ov::AnyMap& properties) override;
+    ov::Any get_property(const std::string& name) const override;
+
+    std::shared_ptr<ov::IAsyncInferRequest> create_infer_request() const override;
+
+    // FIXME: Publicly available for LLMInferRequest
+    std::shared_ptr<ov::npuw::CompiledModel> kvcache_compiled;
+    std::shared_ptr<ov::npuw::CompiledModel> prefill_compiled;
```

Comment:
No need to make it public, you can make friends instead, we already do. :)

Reply:
Will do!

Reply:
Fixed